File: moderlv2.Server/Models/JournalEntry.cs
namespace moderlv2.Server.Models
{
    public class JournalEntry
    {
        public int Id { get; set; }
        public string UserId { get; set; }
        public DateTime Date { get; set; }
        public Mood MoodRating { get; set; }
        public string Description { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        public User User { get; set; }
    }
}

File: moderlv2.client/src/app/models/JournalEntry.ts
export enum Mood {
  VeryBad = 0,
  Bad = 1,
  Neutral = 2,
  Good = 3,
  VeryGood = 4,
  Excellent = 5,
  Amazing = 6
}

export interface JournalEntry {
  id?: number;
  userId: string;
  date: string;
  moodRating: Mood;
  description: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateJournalEntry {
  userId: string;
  date: string;
  moodRating: Mood;
  description: string;
}
